web3/contract.py

```diff
@@ -106,6 +107,19 @@ def __init__(self, abi, web3, address=None):
                         address=address,
                         function_identifier=func['name']))
 
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        self._iter_current_func += 1
```

Comment:
This stateful implementation isn't going to work for us.

I believe the following should work.

```python
def __iter__(self):
    for func in self._functions:
        yield func['name']
```